test/view/generate_view.cpp

```diff
@@ -0,0 +1,132 @@
+#include <stl2/view/generate_view.hpp>
+
+#include "../simple_test.hpp"
+#include <cstring>
+#include <stl2/ranges.hpp>
+#include <stl2/view/take_exactly.hpp>
+
+namespace ranges = __stl2;
+
+namespace view {
+	using namespace ranges::view;
+	using namespace ext;
+} // namespace view
+
+// a simple type to test move semantics
+struct MoveOnlyString {
+	char const* sz_;
+
+	MoveOnlyString(char const* sz = "")
+	  : sz_(sz)
+	{}
+	MoveOnlyString(MoveOnlyString&& that)
+	  : sz_(that.sz_)
+	{
+		that.sz_ = "";
+	}
+	MoveOnlyString(MoveOnlyString const&) = delete;
+	MoveOnlyString& operator=(MoveOnlyString&& that)
+	{
+		sz_ = that.sz_;
+		that.sz_ = "";
+		return *this;
+	}
+	MoveOnlyString& operator=(MoveOnlyString const&) = delete;
+	bool operator==(MoveOnlyString const& that) const
+	{  return 0 == std::strcmp(sz_, that.sz_); }
+
+	bool operator<(const MoveOnlyString& that) const
+	{ return std::strcmp(sz_, that.sz_) < 0; }
+
+	bool operator!=(MoveOnlyString const& that) const
+	{ return !(*this == that); }
+
+	friend std::ostream& operator<<(std::ostream& sout, MoveOnlyString const& str)
+	{ return sout << '"' << str.sz_ << '"'; }
+};
+
+struct MoveOnlyFunction {
+	MoveOnlyString str_;
+	int i_;
+
+	char operator()()
+	{ return str_.sz_[i_++]; }
+};
+
+int main()
+{
+	// Test for constant generator functions
+	{
+		int i = 0, j = 1;
+		auto fib = view::generate([&]()->int{int tmp = i; i += j; std::swap(i, j); return tmp;});
+		static_assert(ranges::InputRange<decltype(fib)>);
+		static_assert(ranges::View<decltype(fib)>);
+		CHECK_EQUAL(fib | view::take_exactly(10), {0,1,1,2,3,5,8,13,21,34});
+	}
+
+	// Test for mutable-only generator functions
+	{
+		int i = 0, j = 1;
+		auto fib = view::generate([=]()mutable->int{int tmp = i; i += j; std::swap(i, j); return tmp;});
+		static_assert(ranges::InputRange<decltype(fib)>);
+		static_assert(ranges::View<decltype(fib)>);
+		CHECK_EQUAL(fib | view::take_exactly(10), {0,1,1,2,3,5,8,13,21,34});
+		// The generator cannot be called when it's const-qualified, so "fib const"
+		// does not model View.
+		static_assert(!ranges::View<decltype(fib) const>);
+	}
+
+	// Test for move-only generator functions
```

Comment:
I'd prefer wrapping this block with `#if 0 // cmcstl2 doesn't support move-only input views` to commenting it out.